harness/determined/cli/workspace.py

```diff
@@ -90,7 +90,7 @@ def render_workspaces(
 def workspace_by_name(sess: api.Session, name: str) -> bindings.v1Workspace:
     w = bindings.get_GetWorkspaces(sess, name=name).workspaces
     if len(w) == 0:
-        raise errors.EmptyResultException(f'Did not find a workspace with name "{name}".')
+        raise cli.not_found_errs("workspace", name, sess)
```

Comment:
there are a set of cli calls that use https://github.com/determined-ai/determined/blob/9a04a46260d75bb566a861561d6458081241d5c6/harness/determined/cli/workspace.py#L53 instead.

this workspace_by_name is a sort of a duplicate that doesn't reuse the logic from that utililty.
could we cover the cases where get_workspace_by_name is also used? 

it might be a good opportunity to write update workspace_by_name to go through the other one or maybe remove one of these.